src/components/profile/ProfileSetProfile.jsx

```diff
@@ -1,20 +1,32 @@
 import PersonIcon from "@mui/icons-material/Person";
 import PhoneIcon from "@mui/icons-material/Phone";
+import PlaceIcon from "@mui/icons-material/Place";
 import { useRequest } from "ahooks";
 import { useForm } from "react-hook-form";
 import { toast } from "react-hot-toast";
 import { useNavigate } from "react-router-dom";
 
-import { useAuth } from "@/models";
+import { useAuth, usePosition } from "@/models";
 
 export default function ProfileSetProfile() {
   const auth = useAuth();
   const navigate = useNavigate();
+  const position = usePosition();
   const { register, handleSubmit } = useForm();
 
   const { run: modifiedContent, loading: modificationLoading } = useRequest(
     async (data) => {
-      return auth.updateProfile(data);
+      return position
+        .getCoordinate(
+          data.address.street,
+          data.address.city,
+          data.address.state,
+          data.address.zipcode
+        )
+        .then((coordinate) => {
```

Comment:
what if this fails?
we should not cease updating the profile even when the coordinate api fails

Reply:
That's the reason why i do catch error in the getCoordinate, it will just return null and everything else works with null coordinate. Users can still browse the page with null coordinate without any problem.